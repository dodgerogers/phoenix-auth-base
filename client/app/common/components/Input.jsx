import React, { PropTypes } from 'react';
import { Form, Label } from 'semantic-ui-react';


const Input = (props) => {
  const hasError = props.meta && props.meta.error && props.meta.touched;

  return (
    <div className="input" style={{ margin: '0 0 1em' }}>
      <Form.Input
        error={hasError}
        {...props}
      />
      {hasError && <Label style={{ marginTop: '-1em' }} basic color="red" pointing>{props.meta.error}</Label>}
    </div>
  )
}

Input.propTypes = {
  input: PropTypes.shape({
    name: PropTypes.string.isRequired,
  }),
  meta: PropTypes.shape({
    touched: PropTypes.bool.isRequired,
    error: PropTypes.string,
  }).isRequired
}

export default Input;
