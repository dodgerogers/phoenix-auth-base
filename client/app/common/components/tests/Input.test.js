import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Input from '../Input';


describe('Input', () => {
  const fullProps = (opts = {}) => ({
    input: {
      name: opts.name || "input-name",
    },
    meta: {
      touched: opts.touched || false,
      error: opts.error || null,
    },
  });

  const pureComponent = (props) => mount(<Input {...props} />);

  it('matches snapshot', () => {
    const component = pureComponent(fullProps());

    expect(toJson(component)).toMatchSnapshot();
  });

  it('matches snapshot when errors are present', () => {
    const props = fullProps({error: 'Something went wrong', touched: true});
    const component = pureComponent(props);

    expect(toJson(component)).toMatchSnapshot();
  });
});
