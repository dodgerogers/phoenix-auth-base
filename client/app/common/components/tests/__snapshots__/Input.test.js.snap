// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input matches snapshot 1`] = `
<Input
  input={
    Object {
      "name": "input-name",
    }
  }
  meta={
    Object {
      "error": null,
      "touched": false,
    }
  }
>
  <div
    className="input"
    style={
      Object {
        "margin": "0 0 1em",
      }
    }
  >
    <FormInput
      as={[Function]}
      control={[Function]}
      error={null}
      input={
        Object {
          "name": "input-name",
        }
      }
    >
      <FormField
        control={[Function]}
        error={null}
        input={
          Object {
            "name": "input-name",
          }
        }
      >
        <div
          className="field"
        >
          <Input
            input={
              Object {
                "name": "input-name",
              }
            }
            type="text"
          >
            <div
              className="ui input"
            >
              <input
                name="input-name"
                onChange={[Function]}
                type="text"
              />
            </div>
          </Input>
        </div>
      </FormField>
    </FormInput>
  </div>
</Input>
`;

exports[`Input matches snapshot when errors are present 1`] = `
<Input
  input={
    Object {
      "name": "input-name",
    }
  }
  meta={
    Object {
      "error": "Something went wrong",
      "touched": true,
    }
  }
>
  <div
    className="input"
    style={
      Object {
        "margin": "0 0 1em",
      }
    }
  >
    <FormInput
      as={[Function]}
      control={[Function]}
      error={true}
      input={
        Object {
          "name": "input-name",
        }
      }
    >
      <FormField
        control={[Function]}
        error={true}
        input={
          Object {
            "name": "input-name",
          }
        }
      >
        <div
          className="error field"
        >
          <Input
            input={
              Object {
                "name": "input-name",
              }
            }
            type="text"
          >
            <div
              className="ui input"
            >
              <input
                name="input-name"
                onChange={[Function]}
                type="text"
              />
            </div>
          </Input>
        </div>
      </FormField>
    </FormInput>
    <Label
      basic={true}
      color="red"
      pointing={true}
      style={
        Object {
          "marginTop": "-1em",
        }
      }
    >
      <div
        className="ui red pointing basic label"
        onClick={[Function]}
        style={
          Object {
            "marginTop": "-1em",
          }
        }
      >
        Something went wrong
      </div>
    </Label>
  </div>
</Input>
`;
